lib/dry/logic/rule.rb

```diff
@@ -47,14 +47,27 @@ def negation
       def new(predicate)
         self.class.new(predicate, options)
       end
-
+      #bit of a hack, essentially genuine proc.curry should be provided it's args via[] whereas ours
+      #accepts them as a method param
       def curry(*args)
-        self.class.new(predicate.curry(*args), options)
+        self.class.new(curried_predicate(*args), options)
       end
 
       def each?
         predicate.is_a?(Rule::Each)
       end
+
+      private
+      def curried_predicate(*args)
+        curry_args = predicate.respond_to?(:arity) && predicate.arity != 0 ? args : []
+        if predicate.is_a?(Proc)
+          predicate.curry[*curry_args]
+        elsif predicate != nil
```

Comment:
this looks suspicious, a rule with a nil predicate means there's a bug in our code, we shouldn't have a guard like that